### PR TITLE
[JSON Schema] Add links to docs

### PR DESCRIPTION
**Pull request by andreafalzetti**
In this PR we introduce links to our docs in the JSON Schema. It couples the links and the JSON Schema but the result is that from the IDE I can jump directly to the Okteto docs ✨ 

Screenshot 2024-12-10 at 15 52 35

The PR also contains minor improvements/fixes (eg. the icon field)

### CODE REVIEW

File: pkg/schema/external.go [hunk 1]
```diff
@@ -57,8 +59,6 @@ func (external) JSONSchema() *jsonschema.Schema {
 
 	return &jsonschema.Schema{
 		Type:                 &jsonschema.Type{Types: []string{"object"}},
-		Title:                "external",
-		Description:          "A list of external resources that are part of your development environment",
```
Comment (ifbyol) [inline pkg/schema/external.go:61]: Just curiosity, why removing the title and description of this item?

File: pkg/schema/icon.go [hunk 2]
```diff
@@ -13,21 +13,14 @@
...
+import (
+	"github.com/kubeark/jsonschema"
+)
 
 type icon struct{}
 
 func (icon) JSONSchema() *jsonschema.Schema {
 	return &jsonschema.Schema{
 		Type: &jsonschema.Type{Types: []string{"string"}},
-		AnyOf: []*jsonschema.Schema{
```
Comment (ifbyol) [inline pkg/schema/icon.go:23]: Why removing this enum? Doesn't it apply anymore?

**Review by ifbyol — APPROVED** [review]
Approving it as this is an improvement, but this is very brittle and hard to maintain. It is really easy that some link is broken and we are not aware. 

We should centralize the links and have a test or step in CI that validates that links are still valid. Also, we have the problem with the versions (of docs and the schema). We will need to think on how to maintain this in a long term

File: pkg/schema/icon.go (hunk 2 above)
Comment (andreafalzetti) [reply pkg/schema/icon.go:23]: The enum should only apply to `external.*.icon`, not the root-level `icon` which is only a string (URL)

File: pkg/schema/external.go (hunk 1 above)
Comment (andreafalzetti) [reply pkg/schema/external.go:61]: The title for root-level properties are defined in the struct with the `jsonschema` annotation, these were getting ignored

**Review by teresaromero — APPROVED** [review]

**Pull request merged**